pkg/apis/batch/v2alpha1/types.go

```diff
@@ -250,6 +250,16 @@ type CronJobSpec struct {
 	// JobTemplate is the object that describes the job that will be created when
 	// executing a CronJob.
 	JobTemplate JobTemplateSpec `json:"jobTemplate" protobuf:"bytes,5,opt,name=jobTemplate"`
+
+	// The number of successful finished jobs to retain.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// +optional
+	SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit,omitempty" protobuf:"varint,6,opt,name=successfulJobsHistoryLimit"`
```

Comment:
> Document the defaults for both of these fields in the godoc.

I haven't seen we document those defaults in the api, elsewhere. 



> Also, what happens when upgrading a cluster? What if someone has failed jobs that they want to keep? Will we suddenly purge them?

It looks like we will, but that's the whole point of introducing these with some reasonable values. We can't keep everything forever.